flappy_bird_example/metal_owl.py

```diff
@@ -0,0 +1,227 @@
+from kivy.clock import Clock # import kivy clock
+from kivy.uix.label import Label # import kivy label
+from kivy.core.audio import SoundLoader # import sound usage
+
+# set the sound source
+sfx_flap = SoundLoader.load("Resources/kivy-game-dev/flappy/audio/flap.wav")
+sfx_score = SoundLoader.load("Resources/kivy-game-dev/flappy/audio/score.wav")
+sfx_die = SoundLoader.load("Resources/kivy-game-dev/flappy/audio/die.wav")
+
+# create the main menu using the widget to add the background, ground, label
+class Menu(Widget):
+    def __init__(self):
+        super(Menu, self).__init__()
+        self.add_widget(Sprite(source="Resources/kivy-game-dev/flappy/images/background.png"))
+        self.size = self.children[0].size
+        self.add_widget(Ground(source="Resources/kivy-game-dev/flappy/images/ground.png"))
+        self.add_widget(Label(center=self.center, text="tap to start"))
+# when click down remove the menu widget and add the game widget as the new parent
+    def on_touch_down(self, *ignore):
+        parent = self.parent
+        parent.remove_widget(self)
+        parent.add_widget(Game())
+
+# when calling sprites use the image import, **kwargs stop from messing with functionality, size is set to the size of the image
+class Sprite(Image):
+    def __init__(self, **kwargs):
+        super(Sprite, self).__init__(**kwargs)
+        self.size = self.texture_size
+
+# the pipe class used as widget, set the top and bottom image as different pipes, add each as widget
+# set top image position to x, with y being 5.5 * the size of the player
+# set the score to false as default
+class Pipe(Widget):
+    def __init__(self, pos):
+        super(Pipe, self).__init__(pos=pos)
+        self.top_image = Sprite(source="Resources/kivy-game-dev/flappy/images/pipe_top.png")
+        self.top_image.pos = (self.x, self.y + 5.5 * 24)
+        self.add_widget(self.top_image)
+        self.bottom_image = Sprite(source="Resources/kivy-game-dev/flappy/images/pipe_bottom.png")
+        self.bottom_image.pos = (self.x, self.y - self.bottom_image.height)
+        self.add_widget(self.bottom_image)
+        self.width = self.top_image.width
+        self.scored = False
+
+# update the pipe widget with the x position moving 2 pipes worth of space
+# set the top and bottom image as the updating self.x value
+# if the pipe is less than 0 position remove the pipe widget
+    def update(self):
+        self.x -= 2
+        self.top_image.x = self.bottom_image.x = self.x
+        if self.right < 0:
+            self.parent.remove_widget(self)
+
+# the pipes class changes the calling of pipes
+# update the pipes by time
+# add a pipe at random height for gap position
+class Pipes(Widget):
+    add_pipe = 0
+    def update(self, dt):
+        for child in list(self.children):
+            child.update()
+        self.add_pipe -= dt
+        if self.add_pipe < 0:
+            y = random.randint(self.y + 50, self.height - 50 - 5.5 * 24)
```

Comment:
Maybe make these numbers (50, 5.5, 24) constants so that it is clear what they represent.